pkg/psmdb/mongo/mongo.go

```diff
@@ -487,15 +492,19 @@ func (m *ConfigMembers) SetVotes() {
 			continue
 		}
 
-		if votes < MaxVotingMembers {
+		// At least one member with data have a vote
+		if member.ArbiterOnly && votes < MaxVotingMembers-1 {
+			// Arbiter should always have a vote and priority=0
 			[]ConfigMember(*m)[i].Votes = 1
+			[]ConfigMember(*m)[i].Priority = 0
 			votes++
-			if !member.ArbiterOnly {
-				[]ConfigMember(*m)[i].Priority = 1
-			}
-		} else if member.ArbiterOnly {
-			// Arbiter should always have a vote
+
+			continue
+		}
+
+		if votes < MaxVotingMembers {
 			[]ConfigMember(*m)[i].Votes = 1
+			[]ConfigMember(*m)[i].Priority = 1
 			votes++
 		}
```

Comment:
I think should we also set explicetely `Votes=0` and `Priority=0` to members if `spec.replsets[].size` > 7 
```suggestion
		} else {
			[]ConfigMember(*m)[i].Votes = 0
 			[]ConfigMember(*m)[i].Priority = 0
 		}
```